core/src/main/java/org/ldaptive/auth/AuthenticationResponse.java

```diff
@@ -50,6 +50,7 @@ public AuthenticationResponse(
     authenticationHandlerResponse = response;
     resolvedDn = dn;
     ldapEntry = entry;
+    ldapEntry.freeze();
```

Comment:
I appreciate this change and think it's a meaningful move toward immutability for authentication-related data. Ideally all data in the authentication response, including the account state, would be immutable.

Reply:
I have an idea to deal with that in a subsequent patch.

Reply:
Sounds good.